### DISCUSSION TITLE
InsertWithIdentity returns DBNull

### DISCUSSION THREAD

**Discussion by ndc** [linq2db/linq2db#4589] [Q&A]
Under what conditions that InsertWithIdentity returns DBNull?
Watching the output window LinqToDB sent this kind of command to the SQL Server:

``` sql server
DECLARE @something NVarChar(4000) -- String
SET     @something = N'...'

INSERT INTO [dbo].[mytable]
(
	[mycolumn]
)
VALUES
(
	@something
)

SELECT SCOPE_IDENTITY()
```

So looks like the SCOPE_IDENTITY retuned null.
After googling this condition [maybe this is a bug in SQL Server](https://stackoverflow.com/questions/2389354/how-can-scope-identity-return-null-when-identity-does-not)?

So for now I'm thinking about getting the id myself: `var identity = db.Query<int?>("select @@identity").First();`

Is there a better way to do that?

**Comment by MaceWindu** [comment]
```
The SCOPE_IDENTITY() function returns the null value if the function is invoked before any INSERT statements into an identity column occur in the scope.
```

e.g. your table doesn't actually have identity column
